src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/AssessmentTest.kt

```diff
@@ -133,6 +134,70 @@ class AssessmentTest : IntegrationTestBase() {
     }
   }
 
+  @Test
+  fun `Get all assessments returns successfully when an inmate details cache failure occurs`() {
+    `Given a User` { user, jwt ->
+      `Given an Offender` { offenderDetails, inmateDetails ->
+        val applicationSchema = approvedPremisesApplicationJsonSchemaEntityFactory.produceAndPersist {
+          withPermissiveSchema()
+        }
+
+        val assessmentSchema = approvedPremisesAssessmentJsonSchemaEntityFactory.produceAndPersist {
+          withPermissiveSchema()
+          withAddedAt(OffsetDateTime.now())
+        }
+
+        val application = approvedPremisesApplicationEntityFactory.produceAndPersist {
+          withCrn(offenderDetails.otherIds.crn)
+          withCreatedByUser(user)
+          withApplicationSchema(applicationSchema)
+        }
+
+        val assessment = approvedPremisesAssessmentEntityFactory.produceAndPersist {
+          withAllocatedToUser(user)
+          withApplication(application)
+          withAssessmentSchema(assessmentSchema)
+          withDecision(null)
+        }
+
+        assessment.schemaUpToDate = true
+
+        val reallocatedAssessment = approvedPremisesAssessmentEntityFactory.produceAndPersist {
+          withAllocatedToUser(user)
+          withApplication(application)
+          withAssessmentSchema(assessmentSchema)
+          withReallocatedAt(OffsetDateTime.now())
+        }
+
+        reallocatedAssessment.schemaUpToDate = true
+
+        // Simulate https://ministryofjustice.sentry.io/issues/4479884804 by deleting the data key from the cache while
+        // preserving the metadata key.
+        val cacheKeys = CacheKeySet(preemptiveCacheKeyPrefix, "inmateDetails", inmateDetails.offenderNo)
+        redisTemplate.delete(cacheKeys.dataKey)
+
+        webTestClient.get()
+          .uri("/assessments")
+          .header("Authorization", "Bearer $jwt")
+          .header("X-Service-Name", ServiceName.approvedPremises.value)
```

Comment:
Curious as to how this got fixed. Presumably this test doesn't start failing if we use temporaryAccommodation here.

Reply:
Maybe we can try both with a `ParameterizedTest`?